handler/create_user.go

```diff
@@ -0,0 +1,65 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"os"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+)
+
+func CreateUser(w http.ResponseWriter, r *http.Request) {
+	db, err := sqlx.Connect("postgres", os.Getenv("POSTGRES_DSN"))
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(fmt.Sprintf("db connection error: %s", err.Error())))
+		return
+	}
+
+	err = r.ParseForm()
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("parse form error: %s", err.Error())))
+		return
+	}
+
+	userId := r.FormValue("user_id")
+	if userId == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("error: missing user_id"))
+		return
+	}
+
+	tx := db.MustBegin()
```

Comment:
i think you need to defer cancelling tx right after this line, or else you risk having orphaned transactions, which results in resource leakage on the database